keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/Resource.kt

```diff
@@ -14,6 +14,25 @@ data class Resource<T : ResourceSpec>(
     require(metadata["application"].isValidApplication()) { "application must be a valid application" }
   }
 
+  /**
+   * Attempts to find an artifact in the delivery config based on information in this resource's spec.
+   */
+  fun findAssociatedArtifact(deliveryConfig: DeliveryConfig) =
```

Comment:
This gets called when generating resource summaries from the API, and delegates to the functions in `DeliveryConfig`, above, to find a matching artifact for the resource (if any) using the image/container provider info.